docs/flex.md

```diff
@@ -0,0 +1,239 @@
+
+# The Flex Backend
+
+**The Flex Backend is experimental. Everything in here is subject to change.**
+
+The "Flex" backend, as the name suggests, allows for a more flexible
+configuration that tells osm2pgsql what OSM data to store in your database and
+exactly where and how. It is configured through a Lua file which
+
+* defines the structure of the output tables and
+* defines functions to map the OSM data to the database data format
+
+See also the example config files in the `flex-config` directory which contains
+lots of comments to get you started.
+
+## The Lua config file
+
+All configuration is done through the `osm2pgsql` object in Lua. It has the
+following fields:
+
+* `osm2pgsql.version`: The version of osm2pgsql as a string.
```

Comment:
I have a question in the context of https://github.com/openstreetmap/osm2pgsql/pull/1064 : 

`osm2pgsql.version` calls `get_osm2pgsql_version()` , which in turn returns a string like "1.2.0 (1.2.0-239-gcc9748f)". If the idea was to enable certain features in a lua script depending on the osm2pgsql version, it might be easier to have `version` return a simple string like "1.2.0" and maybe have an additional `fullversion` attribute returning the full version.

Reply:
I have changed this to return the released version number only.